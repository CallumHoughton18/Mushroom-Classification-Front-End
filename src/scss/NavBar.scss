@import "./Theme.scss";

$navbar-height: 60px;
$navbar-margin-bottom: 10;
$max-width-for-hamburger: 600px;

.navbar {
    font-family: $font-stack;
    min-height: $navbar-height;
    height: $navbar-height;
    background: $secondary-color;
    width: 100%;
    transition: height 0.2s;
    -webkit-box-shadow: 0 8px 8px -7px rgba(0, 0, 0, 0.5);
    -moz-box-shadow: 0 8px 8px -7px rgba(0, 0, 0, 0.5);
    box-shadow: 0 8px 8px -7px rgba(0, 0, 0, 0.5);
    position: -webkit-sticky; /* Safari */
    position: sticky;
    top: 0;

    li a {
        display: inline-block;
        text-align: center;
        padding: 10px 10px;
        text-decoration: none;
        font-weight: bold;
        color: $primary-color-highlight;
        border-radius: 3px;
    }

    li {
        margin-left: 10px;
        margin-right: 10px;
    }

    li a:hover {
        background-color: $secondary-color-lighter;
        transition: 0.2s;
    }

    ul {
        height: 100%;
        display: flex;
        align-items: center;
        flex-direction: row;
        list-style-type: none;
        margin: 0 20px;
        padding: 0;
        overflow: hidden;
        float: right;
    }

    .hamburger {
        font-size: 30px;
        color: $primary-color-highlight;
        cursor: pointer;
        display: none;
    }

    .logo {
        float: left;
        max-height: $navbar-height;
        scale: 0.8;
    }

    @media (max-width: $max-width-for-hamburger) {
        .hamburger {
            display: block;
            float: left;
            margin: 14px 10px;
        }

        ul {
            display: none;
        }

        .logo {
            float: right;
            margin: 0 2px;
        }
    }
}

.expanded {
    @media (max-width: $max-width-for-hamburger) {
        height: 170px;

        ul {
            border-top: 2px solid $primary-color-highlight;
            margin: 0 5px;
            clear: both;
            float: none;
            height: 170px - $navbar-height;
            display: flex;
            align-items: center;
            flex-direction: column;
        }

        li {
            margin-top: 10px;
        }

        li a {
            padding: 8px 30px;
        }
    }
}
