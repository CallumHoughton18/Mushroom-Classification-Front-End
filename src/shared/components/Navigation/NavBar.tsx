import React, {FunctionComponent} from "react";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faBars} from "@fortawesome/free-solid-svg-icons";

import NavLinks from "./NavLinks";
import logo from "../../../icons/logo.png";
import useNavbarExpanded from "../../hooks/useNavbarExpanded";

import sassVars from "../../../stylesheets/abstractions/_variables.scss";

const NavBar: FunctionComponent = () => {
    //TODO: Add type to navStyle? currently just imported as module alias
    const widthForHamburger = parseInt(sassVars.maxWidthForHamburger, 10);
    console.log(widthForHamburger);
    const [expanded, setExpanded] = useNavbarExpanded(widthForHamburger, "resize");
    return (
        <div className={expanded ? "navbar expanded" : "navbar"}>
            <img src={logo} className="logo" />
            <FontAwesomeIcon
                icon={faBars}
                onClick={() => {
                    setExpanded(!expanded);
                }}
                className="hamburger"
            />
            <NavLinks
                onLinkClicked={() => {
                    setExpanded(false);
                }}
            />
        </div>
    );
};

export default NavBar;
