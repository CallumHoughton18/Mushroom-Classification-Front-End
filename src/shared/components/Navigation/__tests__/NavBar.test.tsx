import React from "react";
import {render} from "@testing-library/react";
import NavBar from "../NavBar";
import useNavBarExpanded from "../../../hooks/useNavbarExpanded";
import {mocked} from "ts-jest/utils";
import {StaticRouter} from "react-router-dom";

jest.mock("../../../hooks/useNavbarExpanded");

describe("<NavBar/> render tests", () => {
    const renderAndAssertOnSnapshot = () => {
        //have to render using StaticRouter as NavBar component uses NavLink components
        const component = render(
            <StaticRouter location="someLocation">
                <NavBar />
            </StaticRouter>
        );
        expect(component).toMatchSnapshot();
    };

    describe("Expanded render tests", () => {
        beforeAll(() => {
            const useNavBarExpandedMock = mocked(useNavBarExpanded, true);
            useNavBarExpandedMock.mockReturnValue([true, jest.fn()]);
        });
        it("should render NavBar with expanded CSS class", () => {
            renderAndAssertOnSnapshot();
        });
    });

    describe("Collapsed render tests", () => {
        beforeAll(() => {
            const useNavBarExpandedMock = mocked(useNavBarExpanded, true);
            useNavBarExpandedMock.mockReturnValue([false, jest.fn()]);
        });
        it("should render NavBar without expanded CSS class", () => {
            renderAndAssertOnSnapshot();
        });
    });
});
