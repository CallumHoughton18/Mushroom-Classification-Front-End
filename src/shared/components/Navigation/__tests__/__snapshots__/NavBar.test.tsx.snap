// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> render tests Collapsed render tests should render NavBar without expanded CSS class 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar"
      >
        <img
          class="logo"
          src="[object Object]"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-w-14 hamburger"
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
        <ul>
          <li>
            <a
              href="/Classification"
            >
              Classification
            </a>
          </li>
          <li>
            <a
              href="/About"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar"
    >
      <img
        class="logo"
        src="[object Object]"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 hamburger"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
      <ul>
        <li>
          <a
            href="/Classification"
          >
            Classification
          </a>
        </li>
        <li>
          <a
            href="/About"
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<NavBar/> render tests Expanded render tests should render NavBar with expanded CSS class 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar expanded"
      >
        <img
          class="logo"
          src="[object Object]"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-w-14 hamburger"
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
        <ul>
          <li>
            <a
              href="/Classification"
            >
              Classification
            </a>
          </li>
          <li>
            <a
              href="/About"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar expanded"
    >
      <img
        class="logo"
        src="[object Object]"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 hamburger"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
      <ul>
        <li>
          <a
            href="/Classification"
          >
            Classification
          </a>
        </li>
        <li>
          <a
            href="/About"
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
